### ISSUE TITLE
Support for SemVer 2.0.0 Packages

### ISSUE THREAD

**Issue by jwittner** [GlitchEnzo/NuGetForUnity#273]
## Description
Right now if packages use the features of SemVer 2.0.0, e.g. +metadata or '.' in the prerelease string, then they won't show up in NuGetForUnity.

To enable this we need to add the `semVerLevel=2.0.0` query arg to our urls, and modify the version comparison logic to support all three version types supported by nuget, i.e. System.Version, SemVer 1.0.0 and Semver 2.0.0.

Relevant Docs:
* [NuGet Docs - Search Query Service](https://docs.microsoft.com/en-us/nuget/api/search-query-service-resource)
* [NuGet GitHub Wiki - SemVer 2.0.0 Support](https://github.com/NuGet/Home/wiki/SemVer2-support-for-nuget.org-%28server-side%29#identifying-semver-v200-packages)

**Comment by derekfreed** [comment]
This would be really useful for preview packages that have versionings such as 1.0.0-preview.2

**Comment by bricelam** [comment]
Superseded by #353?

**Comment by jwittner** [comment]
@bricelam - unrelated to the change in API endpoints.

**Comment by Henri-J-Norden** [comment]
I needed a package ([Colourful 3.1.0](https://www.nuget.org/packages/Colourful)), where without SemVer 2 only version [2.0.5](https://www.nuget.org/packages/Colourful/2.0.5) is available. I don't know much about nuget, but I made this workaround so at least SemVer 2 packages can be added manually into `Assets/packages.config` like this: 
```xml
<packages>
  <package id="Colourful" version="3.1.0" />
</packages>
```

It will then fetch the package when you click `NuGet->Restore Packages` in Unity. Though the package management GUI still shows that the latest version is 2.0.5, so automatic updates won't work and it's not really a solution to this issue... but in case someone else needs such a workaround:

Git url for Unity package manager: 
```
https://github.com/norden-io/NuGetForUnity.git?path=/src/NuGetForUnity#SemVer2
```
Source: https://github.com/norden-io/NuGetForUnity/tree/SemVer2

**Comment by JoC0de** [comment]
fixed by #543
